### PR TITLE
Add merge metric

### PR DESCRIPTION
**Pull request by millb**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- New Feature

Short description (up to few sentences):
New ProfileEvent `Merge` added. It counts the number of launched background merges.

### CODE REVIEW

File: dbms/src/Common/ProfileEvents.cpp [hunk 1]
```diff
@@ -114,6 +114,7 @@
     M(SelectedRanges, "Number of (non-adjacent) ranges in all data parts selected to read from a MergeTree table.") \
     M(SelectedMarks, "Number of marks (index granules) selected to read from a MergeTree table.") \
     \
+    M(CountOfMerges, "") \
```
Comment (alesapin) [inline dbms/src/Common/ProfileEvents.cpp]: Need description.

Comment (alexey-milovidov) [reply dbms/src/Common/ProfileEvents.cpp]: We don't have `Count` in the names of other metrics because counting is assumed.

Comment (alexey-milovidov) [reply dbms/src/Common/ProfileEvents.cpp]: Let it be named `Merge`.

**Comment by alexey-milovidov** [comment]
Need to add a test that the value of this ProfileEvent is greater than zero after merge.

**Comment by alexey-milovidov** [comment]
You should fill the description for changelog.

**Comment by hustnn** [comment]
Hi @alexey-milovidov 

I want to add elapsed time for each stream like **BlockInputStream to get more detailed performance result? Is there any example on it? 

Thanks.

**Pull request merged**